src/unix/unix_internal.h

```diff
@@ -885,17 +889,26 @@ static inline boolean validate_process_memory(process p, const void *a, bytes le
 
 static inline boolean thread_in_interruptible_sleep(thread t)
 {
-    return t->blocked_on && t->blocked_on != INVALID_ADDRESS;
+    unix_context ctx = &t->syscall->uc;
+    if (!ctx)
```

Comment:
Yes, this is meant to check if t->syscall is null, but since the code is only evaluating the address of t->syscall->uc, there is no actual dereference in the assignment of the ctx variable.
Anyway, this code would stop working if `uc` wasn't the first field of the syscall_context struct; for better clarity, I changed it (and the following functions as well) to explicitly check for a null t->syscall.